From 9eb311e16242c03d770cc47c409e9a679ba57171 Mon Sep 17 00:00:00 2001
From: Bharathraj Nagaraju <bnagar@motorola.com>
Date: Fri, 10 Apr 2015 03:42:32 -0500
Subject: [PATCH 238/959] IKSWL-7300: power: smb135x: Map COOL & WARM health
 states to GOOD.

Since Android does not support COOL or WARM, map them to GOOD state.

Change-Id: I0e43a44e5b9e015efe2c663170c6463e984e4813
Signed-off-by: Bharathraj Nagaraju <bnagar@motorola.com>
Reviewed-on: http://gerrit.mot.com/733698
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Kenneth Kessler <kennykessler@motorola.com>
Reviewed-by: Jason Knopsnyder <jasonk@motorola.com>
Reviewed-by: Vidyadhar Gurram <rnh786@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/power/qpnp-smbcharger-mmi.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/power/qpnp-smbcharger-mmi.c b/drivers/power/qpnp-smbcharger-mmi.c
index 0395eb86bf8..ff6f5338c10 100644
--- a/drivers/power/qpnp-smbcharger-mmi.c
+++ b/drivers/power/qpnp-smbcharger-mmi.c
@@ -2757,6 +2757,19 @@ static int smbchg_battery_get_property(struct power_supply *psy,
 	case POWER_SUPPLY_PROP_HEALTH:
 		val->intval = get_prop_batt_health(chip);
 		val->intval = chip->temp_state;
+		if (val->intval ==  POWER_SUPPLY_HEALTH_WARM) {
+			if (chip->ext_high_temp)
+				val->intval = POWER_SUPPLY_HEALTH_OVERHEAT;
+			else
+				val->intval = POWER_SUPPLY_HEALTH_GOOD;
+		}
+
+		if (val->intval ==  POWER_SUPPLY_HEALTH_COOL) {
+			if (chip->ext_high_temp)
+				val->intval = POWER_SUPPLY_HEALTH_COLD;
+			else
+				val->intval = POWER_SUPPLY_HEALTH_GOOD;
+		}
 		break;
 	case POWER_SUPPLY_PROP_TECHNOLOGY:
 		val->intval = POWER_SUPPLY_TECHNOLOGY_LION;
-- 
2.11.0

